### PR TITLE
clone settings construction

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: src/app/integrations/xero/xero-main/xero-dashboard/xero-dashboard.component.ts [hunk 1]
```diff
@@ -145,8 +145,8 @@ export class XeroDashboardComponent implements OnInit {
       this.xeroExportSettingService.getExportSettings()
     ]).subscribe((responses) => {
       this.errors = DashboardModel.parseAPIResponseToGroupedError(responses[0]);
-      this.reimbursableImportState = responses[4].workspace_general_settings?.reimbursable_expenses_object ? this.reimbursableExpenseImportStateMap[responses[4].expense_group_settings.expense_state] : null;
-      this.cccImportState = responses[4].workspace_general_settings?.corporate_credit_card_expenses_object ? this.cccExpenseImportStateMap[responses[4].expense_group_settings.ccc_expense_state] : null;
+      this.reimbursableImportState = responses[4].workspace_general_settings?.reimbursable_expenses_object ? this.reimbursableExpenseImportStateMap[ExpenseState.PAID] : null;
+      this.cccImportState = responses[4].workspace_general_settings?.corporate_credit_card_expenses_object ? this.cccExpenseImportStateMap[CCCExpenseState.APPROVED] : null;
```
Comment (ashwin1111) [inline src/app/integrations/xero/xero-main/xero-dashboard/xero-dashboard.component.ts:149]: why are these hard coded?

File: src/app/integrations/xero/xero-onboarding/xero-clone-settings/xero-clone-settings.component.html [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+                    [formControllerName]="'changeAccountingPeriod'"
+                    [inputType]="InputType.TOGGLE"
+                    [tooltipText]="'If the accounting period in Xero is closed, the expenses from ' + brandingConfig.brandName + ' will be exported with a date stamp of the first day next open accounting period.'">
+                </app-clone-setting-field>
+            </div>
+
+            <div class="clone-setting--field">
+                <app-clone-setting-field
+                    [iconSource]="'user-plus'"
+                    [label]="'Auto-Create Vendors'"
```
Comment (ashwin1111) [inline src/app/integrations/xero/xero-onboarding/xero-clone-settings/xero-clone-settings.component.html:332]: Contacts

File: src/app/integrations/xero/xero-onboarding/xero-clone-settings/xero-clone-settings.component.html [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+                    [iconSource]="'calendar'"
+                    [label]="'Post entries in the next open accounting period'"
+                    [form]="advancedSettingForm"
+                    [formControllerName]="'changeAccountingPeriod'"
+                    [inputType]="InputType.TOGGLE"
+                    [tooltipText]="'If the accounting period in Xero is closed, the expenses from ' + brandingConfig.brandName + ' will be exported with a date stamp of the first day next open accounting period.'">
+                </app-clone-setting-field>
+            </div>
+
+            <div class="clone-setting--field">
```
Comment (ashwin1111) [inline src/app/integrations/xero/xero-onboarding/xero-clone-settings/xero-clone-settings.component.html:329]: this should be visible only when reimbursable export is enabled

File: src/app/integrations/xero/xero-main/xero-dashboard/xero-dashboard.component.ts (hunk 1 above)
Comment (DhaaraniCIT) [reply src/app/integrations/xero/xero-main/xero-dashboard/xero-dashboard.component.ts:149]: it is place i changed to resolve the error I got in local, reverted it

**Review by ashwin1111 — APPROVED** [review]

**Comment by DhaaraniCIT** [comment]
Screenshot 2024-03-18 at 8 41 47 PM
Screenshot 2024-03-18 at 8 41 55 PM
Screenshot 2024-03-18 at 8 42 05 PM

**Pull request merged**